test/e2e/generator/generate.go

```diff
@@ -208,12 +212,21 @@ func generateTestnet(r *rand.Rand, opt map[string]interface{}) (e2e.Manifest, er
 			node.UseLegacyP2P = true
 		case HybridP2PMode:
 			node.UseLegacyP2P = r.Float64() < legacyP2PFactor
-
 			if node.UseLegacyP2P {
-				hasHybridValidator = true
-			} else if !hasHybridValidator {
-				hasHybridValidator = true
-				node.UseLegacyP2P = true
+				hybridNumLegacy++
+				if hybridNumNew == 0 {
+					hybridNumNew++
```

Comment:
I am not entirely sure what this is trying to do. I thought the point was to count how many we have of each, but the deductions make me think this is a balancing test? If that's true, though, can't we just trust the random factor?

Reply:
That's a good point. I wonder then if we shouldn't take randomness out entirely, and just use the cumulative distribution to assign the fraction we expect, possibly with a minimum case, e.g., https://play.golang.org/p/Gp4Wn_0xQtr.

That's a little more setup, but would ensure we get the right fraction (modulo bucket sizes), even when dice don't work. YMMV.

Reply:
The hybrid node stuff all disappears when we delete the legacy P2P code, so I'm less inclined to fuss over it at this point.